default/python/AI/ShipDesignAI.py

```diff
@@ -372,17 +370,15 @@ def _check_cache_for_consistency(self):
             try:
                 cached_name = fo.getShipDesign(self.design_id_by_name[designname]).name
                 if cached_name != designname:
-                    print "  WARNING: ShipID cache corrupted."
-                    print "    Expected: %s, got: %s. Repairing cache." % (designname, cached_name)
+                    warn("ShipID cache corrupted. Expected: %s, got: %s." % (designname, cached_name))
                     design_id = next(iter([shipDesignID for shipDesignID in fo.getEmpire().allShipDesigns
                                           if designname == fo.getShipDesign(shipDesignID).name]), None)
                     if design_id is not None:
                         self.design_id_by_name[designname] = design_id
                     else:
                         corrupted.append(designname)
             except AttributeError:
-                print "  WARNING: ShipID cache corrupted. Could not get cached shipdesign. Repairing Cache."
-                print traceback.format_exc()  # do not print to stderr as this is an "expected" exception.
+                warn("ShipID cache corrupted. Could not get cached shipdesign. Repairing Cache.", exc_info=True)
```

Comment:
Should this be an error?  Is it possible for the cache to be corrupted and still operating correctly?

Reply:
The history of this is that some design ids somehow magically changed during turns once (probably related to a moderator ingame). Within a single turn, the cache worked fine. Next turn, there was a mismatch between cached and server ids. So, yes, the cache may very well work fine (though inefficient) if such a mismatch happens.
If there is intra-turn mismatch between the data, there is no way the AI can play anyway as it would get "random" data from the server and pretty much all the code is unlikely to work at that point.